### PR TITLE
Added mysql config options and example config.

### PR DESCRIPTION
**Pull request by jason-o-matic**

### CODE REVIEW

File: lib/telegraf/telegraf.conf.erb [hunk 1]
```diff
@@ -572,7 +573,9 @@
   ##    root@tcp(127.0.0.1:3306)/?tls=false
   #
   ## If no servers are specified, then localhost is used as the host.
-  servers = ["root@tcp(127.0.0.1:3306)/"]
+  servers = <%= config_file['mysql'] %>
+  tagexclude = ["server"]
```
Comment (mediocretes) [inline lib/telegraf/telegraf.conf.erb]: I actually think we will want this in, or we will want to be sure we're setting some root namespace explicitly.  the reported name of ip:port is not as good as a real name, but I think it's something people will want in their metrics - the name of the server you're talking to is more useful than the name of the server you're talking from.

File: lib/telegraf/telegraf.conf.erb [hunk 2]
```diff
@@ -616,6 +619,7 @@
   #
   ## Some queries we may want to run less often (such as SHOW GLOBAL VARIABLES)
   interval_slow                   = "30m"
```
Comment (mediocretes) [inline lib/telegraf/telegraf.conf.erb]: Do we know if any of the things we're passing are in the slow category?  I don't think so, but if they are, we may need to make this interval smaller.

**Comment by mediocretes** [comment]
Jason, you are a hero.

File: lib/telegraf/telegraf.conf.erb (hunk 2 above)
Comment (jason-o-matic) [inline lib/telegraf/telegraf.conf.erb]: I don't think so, I didn't see anything other than the obvious `fieldpass` metrics show up in my testing.

File: lib/telegraf/telegraf.conf.erb (hunk 1 above)
Comment (jason-o-matic) [inline lib/telegraf/telegraf.conf.erb]: I agree. I'll post in the basecamp todo asking about metric naming here and update it once we decide.

Comment (mediocretes) [inline lib/telegraf/telegraf.conf.erb]: :+1: I just saw that thread, agree.

File: lib/telegraf/telegraf.conf.erb (hunk 2 above)
Comment (jason-o-matic) [inline lib/telegraf/telegraf.conf.erb]: Actually, I just noticed this in some testing `mysql_users.127_0_0_1.root.connections` so we might need to filter that out.

**Comment by jason-o-matic** [comment]
I've updated this so we get metric names like `mysql.127_0_0_1.bytes_received`. I think the strategy I've taken is basically how we'll want to do this for other services. It's not the most beautiful, but it's not too bad.

File: lib/telegraf/telegraf.conf.erb (hunk 2 above)
Comment (jason-o-matic) [inline lib/telegraf/telegraf.conf.erb]: Done with `namedrop`.

**Comment by janxious** [comment]
For all the options in the mysql settings that are enabled but being filtered out, should we just disable  them?

**Comment by janxious** [comment]
I take it mysql doesn't report metrics by the database, but by the server instance?

**Comment by jason-o-matic** [comment]
> For all the options in the mysql settings that are enabled but being filtered out, should we just disable them?

I just committed https://github.com/Instrumental/instrumentald/pull/3/commits/6edc433bd2a1d968685b65ab2e3191b7f979a62e which I think is what you're talking about, let me know otherwise. I think that's the most we can do to disable things.


> I take it mysql doesn't report metrics by the database, but by the server instance?

It reports per-user stats, so I assume it does per-db stats, but I haven't actually checked. I thought we decided to just start with the mysql metrics that `instrumental_tools` reported, but I can't remember where I read that so maybe I'm smoking it. If that's not the case we can talk some more in basecamp about what we want to collect.

**Comment by janxious** [comment]
> I just committed 6edc433 
Yup!

Asking about database versus server might be important if any of the metrics is actually reported per DB, though, I wouldn't expect it given mySQL's architecture. `queries` and `slow_queries` make sense as per-db, for example.

File: lib/telegraf/telegraf.conf.erb [hunk 3]
```diff
@@ -73,7 +73,7 @@
   prefix = ""
   ## Stats output template (Graphite formatting)
   ## see https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md#graphite
-  template = "host.tags.measurement.field"
+  template = "host.measurement.tags.field"
```
Comment (janxious) [inline lib/telegraf/telegraf.conf.erb]: Once this is merged, we need to re-run everything and make sure we're doing for all other metrics.

**Comment by jqr** [comment]
Code looks good 👍

File: lib/telegraf/telegraf.conf.erb (hunk 3 above)
Comment (jason-o-matic) [inline lib/telegraf/telegraf.conf.erb]: > make sure we're doing for
Did you accidentally a word there?

Comment (janxious) [inline lib/telegraf/telegraf.conf.erb]: "doing right"

Comment (jason-o-matic) [inline lib/telegraf/telegraf.conf.erb]: Ah, yep, totally agree.

File: lib/telegraf/telegraf.conf.erb [hunk 4]
```diff
@@ -562,6 +562,7 @@
 #   servers = ["127.0.0.1:27017"]
 
 
+<% config_file['mysql'].each do |server_url| %>
```
Comment (jqr) [inline lib/telegraf/telegraf.conf.erb]: This call to `each` will fail when `config_file['mysql']` returns `nil`.

**Pull request merged**